lib/pure/asyncmacro.nim

```diff
@@ -440,7 +399,24 @@ macro multisync*(prc: untyped): untyped =
   result.add(asyncSingleProc(asyncPrc))
   result.add(sync)
 
-proc await*[T](x: T) =
-  ## The 'await' keyword is also defined here for technical
-  ## reasons. (Generic symbol lookup prepass.)
-  {.error: "Await only available within .async".}
+# seems simpler to define it here, than to define it in each multisync
+template await*(f: untyped): untyped =
+  when declared(internalRetFuture):
+    error "Await expects a Future"
+  elif not declared(inMultisync):
+    error "Await only available within {.async.}"
+  else:
+    f
+ 
+# based on the yglukhov's patch to chronos: https://github.com/status-im/nim-chronos/pull/47
+template await*[T](f: Future[T]): auto =
```

Comment:
AFAIK we shouldn't use the `auto` return type in the stdlib

Reply:
hm, is there a more complete description why (no problem to fix it)

Reply:
because `auto` sucks for documentation and means that the template/proc can effectively return anything in the future.

In this case the return type should be trivial to specify completely, just use ``T``, no?

Reply:
it doesnt work for some reason with `T` : expr has no type errors, it really doesnt seem obvious to me why

Reply:
ok, as we discussed IIRC it can stay with `auto`